frame/offences/src/lib.rs

```diff
@@ -92,6 +97,21 @@ pub mod pallet {
 		ValueQuery,
 	>;
 
+	/// A vector of active reports sorted by `SessionIndex`.
+	/// For each session index, all reports are stored along with
+	/// their kind and time slot info.
+	///
+	/// On start of a new session, all reports with a session index older than
+	/// `MaxSessionReportAge` are removed.
+	///
+	/// Note that time_slot is encoded as stored as an opaque type.
+	#[pallet::storage]
+	pub type SessionReports<T: Config> = StorageValue<
+		_,
+		Vec<u8>, //(SessionIndex, Vec<(Kind, OpaqueTimeSlot, ReportIdOf<T>)>)
```

Comment:
Why is this not a `Map<SessionIndex, (stuff)>`?

Reply:
We need to keep the items sorted by `SessionIndex` so that we can delete the older ones on the start of a new session. I assume that won't be possible with `Map` but let me know if there's a way.

Reply:
As discussed, added a `Map` as we clean up only one `SessionIndex` on the start of a new session